### DISCUSSION TITLE
Hosting quartz in ASP.NET core and pause the scheduler

### DISCUSSION THREAD

**Discussion by gkarwchan** [quartznet/quartznet#1915] [Q&A]
I want to host quartz.net inside asp.net core.
I am using Quartz.aspnetcore package.

But I am looking for a way to pause the scheduler until I manually start it.

I am doing the following right now

```
builder.Services.AddQuartz(q =>
{
    q.UseMicrosoftDependencyInjectionJobFactory();
});

builder.Services.AddQuartzHostedService(opt =>
{
    opt.WaitForJobsToComplete = true;
});
var factory = app.Services.GetRequiredService<ISchedulerFactory>();
var scheduler = await factory.GetScheduler();

// build all jobs and triggers.....
// after scheduling all jobs do this
scheduler.PauseAll();


```
Is there a better way to do that?
Is there a setting in AddQuartzHostedService to pause the scheduler and not start it?

And another question

what is the difference between : AddQuartzHostedService and AddQuartzServer

**Comment by lahma** [comment]
So your ask is how to prevent Quartz starting as a hosted service?

**Reply by gkarwchan** [reply]
K
Got it. Then I have to not use the hosted service
